spec/licensee/projects/github_project_spec.rb

```diff
@@ -0,0 +1,140 @@
+RSpec.describe Licensee::Projects::GitHubProject do
+  describe '#initialize' do
+    subject(:instance) { described_class.new(github_url) }
+
+    context 'with a GitHub URI' do
+      let(:github_url) { 'https://github.com/benbalter/licensee' }
+
+      it 'should set @repo' do
+        expect(instance.repo).to eq('benbalter/licensee')
+      end
+    end
+
+    context 'with a GitHub git URI' do
+      let(:github_url) { 'https://github.com/benbalter/licensee.git' }
+
+      it 'should set @repo, stripping the trailing extension' do
+        expect(instance.repo).to eq('benbalter/licensee')
+      end
+    end
+
+    context 'with a non-GitHub URI' do
+      let(:github_url) { 'https://gitlab.com/benbalter/licensee' }
+
+      it 'should raise an ArgumentError' do
+        expect { instance }.to raise_error(ArgumentError)
+      end
+    end
+
+    context 'with a local folder' do
+      let(:github_url) { fixture_path('mit') }
+
+      it 'should raise an ArgumentError' do
+        expect { instance }.to raise_error(ArgumentError)
+      end
+    end
+  end
+
+  subject { described_class.new(github_url) }
+
+  let(:repo) { 'benbalter/licensee' }
+  let(:github_url) { 'https://github.com/benbalter/licensee' }
+  let(:mit) { Licensee::License.find('mit') }
+  let(:readme_file) do
+    File.read(fixture_path('mit/README.md'))
+  end
+  let(:license_file) { File.read(fixture_path('mit/LICENSE.txt')) }
+
+  context 'when the repo exists' do
+    before do
+      allow(Octokit)
+        .to receive(:contents)
+        .with('benbalter/licensee')
+        .and_return([
```

Comment:
Would never have thought to do it this way. For other projects, I've used VCR, which stubs the HTTP API response, rather than the library method, in case the library changes its behavior. Out of scope for this PR, and glad to implement as a follow up.